pkg/registry/admissionregistration/mutatingadmissionpolicy/authz.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mutatingadmissionpolicy
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apiserver/pkg/authorization/authorizer"
+	genericapirequest "k8s.io/apiserver/pkg/endpoints/request"
+	"k8s.io/kubernetes/pkg/apis/admissionregistration"
+	rbacregistry "k8s.io/kubernetes/pkg/registry/rbac"
+)
+
+func (v *mutatingAdmissionPolicyStrategy) authorizeCreate(ctx context.Context, obj runtime.Object) error {
+	policy := obj.(*admissionregistration.MutatingAdmissionPolicy)
+	if policy.Spec.ParamKind == nil {
+		// no paramRef in new object
+		return nil
+	}
+
+	return v.authorize(ctx, policy)
+}
+
+func (v *mutatingAdmissionPolicyStrategy) authorizeUpdate(ctx context.Context, obj, old runtime.Object) error {
+	policy := obj.(*admissionregistration.MutatingAdmissionPolicy)
+	if policy.Spec.ParamKind == nil {
+		// no paramRef in new object
+		return nil
+	}
+
+	oldPolicy := old.(*admissionregistration.MutatingAdmissionPolicy)
+	if oldPolicy.Spec.ParamKind != nil && *oldPolicy.Spec.ParamKind == *policy.Spec.ParamKind {
+		// identical paramKind to old object
+		return nil
+	}
+
+	return v.authorize(ctx, policy)
+}
+
+func (v *mutatingAdmissionPolicyStrategy) authorize(ctx context.Context, policy *admissionregistration.MutatingAdmissionPolicy) error {
+	if v.authorizer == nil || policy.Spec.ParamKind == nil {
+		return nil
+	}
+
+	// for superuser, skip all checks
+	if rbacregistry.EscalationAllowed(ctx) {
```

Comment:
This is a little misuse of the method (preexisting in the validating policy case). Wouldn't mind to make it explicit: `isSystemAdmin` or similar.

Reply:
OK if I follow up on this separately since we'll be changing ValidatingAdmissionPolicy as well?  I've opened https://github.com/kubernetes/kubernetes/issues/127935 to track.